xunit-autolabeler-v2/ast_parser/lib/file_utils.py

```diff
@@ -0,0 +1,116 @@
+# Copyright 2020 Google LLC.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+from typing import List, Callable
+
+import re
+import os
+import subprocess
+
+from . import constants
+
+
+def _getFiles(root_dir: str, predicate: Callable[[str], bool]) -> List[str]:
+    """Recursively list the files in a given directory
+       whose names match the provided predicate function
+
+    Args:
+        root_dir: the root directory to search from
+        predicate: the predicate function
+                                       to filter filenames with
+
+    Returns:
+        A list of filepaths relative to root_dir that match the predicate
+    """
+    if os.path.basename(os.path.normpath(root_dir)).startswith('.'):
+        # Ignore dot-directories
+        return []
+
+    paths = [os.path.join(root_dir, path) for path in os.listdir(root_dir)]
+
+    folders = [path for path in paths if not os.path.isfile(path)]
+
+    files = [path for path in paths if os.path.isfile(path)
+             and predicate(path)]
+
+    for file in folders:
+        files += _getFiles(file, predicate)
+
+    return files
+
+
+def get_python_files(root_dir: str) -> List[str]:
+    """Recursively lists the Python files in a directory
+
+    Args:
+        root_dir: the root directory to search from
+
+    Returns:
+        A list of Python filepaths relative to root_dir
+    """
+
+    # Not language-agnostic, so keep it in this method
```

Comment:
If you want to share this with other language case, should we put it outside?

Reply:
Since this is only used in Python (for now at least), I'd rather keep it in this method.

Reply:
Ah right. I misread the comment. nm